src/users/users.service.ts

```
@@ -1,27 +1,24 @@
     return newUser;
   }
 
-  findOne(id: string): User {
-    const user = this.users.find(user => user.id === id);
-    if (!user) {
-      throw new NotFoundException(`User with ID "${id}" not found`);
-    }
-    return user;
+  findOne(id: string): User | undefined {
+    // Using '==' allows for type coercion bugs
+    const user = this.users.find(user => user.id == id);
```

ESLint

[error] Replace `user` with `(user)`

(prettier/prettier)